nimbus-jose-jwt_aws-kms-extension/src/main/java/com/nimbusds/jose/aws/kms/crypto/impl/KmsAsymmetricSigningCryptoProvider.java

```diff
@@ -66,33 +66,40 @@ public abstract class KmsAsymmetricSigningCryptoProvider extends BaseJWSProvider
 
     public static final Map<JWSAlgorithm, String> JWS_ALGORITHM_TO_MESSAGE_DIGEST_ALGORITHM =
             ImmutableMap.<JWSAlgorithm, String>builder()
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256.toString()),
-                            MessageDigestAlgorithms.SHA_256)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_384.toString()),
-                            MessageDigestAlgorithms.SHA_384)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_512.toString()),
-                            MessageDigestAlgorithms.SHA_512)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_256.toString()),
-                            MessageDigestAlgorithms.SHA_256)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_384.toString()),
-                            MessageDigestAlgorithms.SHA_384)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_512.toString()),
-                            MessageDigestAlgorithms.SHA_512)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_256.toString()),
-                            MessageDigestAlgorithms.SHA_256)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_384.toString()),
-                            MessageDigestAlgorithms.SHA_384)
-                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_512.toString()),
-                            MessageDigestAlgorithms.SHA_512)
+                    .put(JWSAlgorithm.RS256, MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.RS384, MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.RS512, MessageDigestAlgorithms.SHA_512)
+                    .put(JWSAlgorithm.PS256, MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.PS384, MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.PS512, MessageDigestAlgorithms.SHA_512)
+                    .put(JWSAlgorithm.ES256, MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.ES384, MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.ES512, MessageDigestAlgorithms.SHA_512)
```

Comment:
This seems to be a backward incompatible change. Can't we retain the support for existing KMS specific algo names, while adding the new RFC compliance algo names?

Reply:
We can, good call - I've pushed an update to include the KMS-specific algo strings into the map so they resolve. Thanks!